solidity/ecdsa/contracts/libraries/EcdsaAuthorization.sol

```diff
@@ -15,25 +15,474 @@
 pragma solidity ^0.8.9;
 
 import "@keep-network/sortition-pools/contracts/SortitionPool.sol";
-import "@threshold-network/solidity-contracts/contracts/staking/TokenStaking.sol";
+import "@threshold-network/solidity-contracts/contracts/staking/IStaking.sol";
 
+/// @notice Library managing the state of stake authorizations for ECDSA
+///         operator contract and the presence of operators in the sortition
+///         pool based on the stake authorized for them.
 library EcdsaAuthorization {
-    struct Data {
+    struct Parameters {
+        // The minimum authorization required by ECDSA application so that
+        // operator can join the sortition pool and do the work.
         uint96 minimumAuthorization;
+        // Authorization decrease delay in seconds between the time
+        // authorization decrease is requested and the time the authorization
+        // decrease can be approved. It is always the same value, no matter if
+        // authorization decrease amount is small, significant, or if it is
+        // a decrease to zero.
         uint64 authorizationDecreaseDelay;
     }
 
+    struct AuthorizationDecrease {
+        uint96 decreasingBy; // amount
+        uint64 decreasingAt; // timestamp
+    }
+
+    struct Data {
+        Parameters parameters;
+        mapping(address => address) stakingProviderToOperator;
+        mapping(address => address) operatorToStakingProvider;
+        mapping(address => AuthorizationDecrease) pendingDecreases;
+    }
+
+    event OperatorRegistered(
+        address indexed stakingProvider,
+        address indexed operator
+    );
+
+    event AuthorizationIncreaseRequested(
```

Comment:
Yeah, good point. I think I named it `*Requested` to reflect the fact the state update is not yet done, and the sortition pool needs to be updated. In retrospect, this is confusing because, for the decrease path, requested+approved means something else. I will rename this event.

Reply:
29b9cb4eec83620c1140db5979aaa480e08644f2